pyqtgraph/jupyter/GraphicsView.py

```diff
@@ -0,0 +1,127 @@
+import jupyter_rfb
+
+__all__ = ['GraphicsLayoutWidget', 'PlotWidget']
+
+KMLUT = {
+    x : getattr(QtCore.Qt.KeyboardModifier, x + "Modifier")
+    for x in ["Shift", "Control", "Alt", "Meta"]
+}
+
+MBLUT = {
+    k : getattr(QtCore.Qt.MouseButton, v + "Button")
+    for (k, v) in zip(
+        range(6),
+        ["No", "Left", "Right", "Middle", "Back", "Forward"]
+    )
+}
+
+TYPLUT = {
+    "pointer_down" : QtCore.QEvent.Type.MouseButtonPress,
+    "pointer_up" : QtCore.QEvent.Type.MouseButtonRelease,
+    "pointer_move" : QtCore.QEvent.Type.MouseMove,
+}
+
+def get_buttons(evt_buttons):
+    NoButton = QtCore.Qt.MouseButton.NoButton
+    btns = NoButton
+    for x in evt_buttons:
+        btns |= MBLUT.get(x, NoButton)
+    return btns
+
+def get_modifiers(evt_modifiers):
+    NoModifier = QtCore.Qt.KeyboardModifier.NoModifier
+    mods = NoModifier
+    for x in evt_modifiers:
+        mods |= KMLUT.get(x, NoModifier)
+    return mods
+
+
+class GraphicsView(jupyter_rfb.RemoteFrameBuffer):
+    def __init__(self, **kwds):
+        super().__init__(**kwds)
+        self.gfxView = widgets.GraphicsView.GraphicsView()
+        self.logical_size = 640, 480
+        self.pixel_ratio = 1.0
+        # self.gfxView.resize(*self.logical_size)
+        # self.gfxView.show()
+        # self.gfxView.resizeEvent(None)
+
+    def get_frame(self):
+        w, h = self.logical_size
+        dpr = self.pixel_ratio
+        buf = np.empty((int(h * dpr), int(w * dpr), 4), dtype=np.uint8)
+        qimg = fn.ndarray_to_qimage(buf, QtGui.QImage.Format.Format_RGBX8888)
+        qimg.fill(QtCore.Qt.GlobalColor.transparent)
+        qimg.setDevicePixelRatio(dpr)
+        painter = QtGui.QPainter(qimg)
+        self.gfxView.render(painter, self.gfxView.viewRect(), self.gfxView.rect())
+        painter.end()
+        return buf
+    
+    def handle_event(self, event):
+        event_type = event["event_type"]
+
+        if event_type == "resize":
+            self.logical_size = int(event["width"]), int(event["height"])
+            self.pixel_ratio = event["pixel_ratio"]
+            self.gfxView.resize(*self.logical_size)
+            self.gfxView.resizeEvent(None)
+        elif event_type in ["pointer_down", "pointer_up", "pointer_move"]:
+            btn = MBLUT.get(event["button"], None)
+            if btn is not None:    # ignore unsupported buttons
+                pos = QtCore.QPointF(event["x"], event["y"])
+                btns = get_buttons(event["buttons"])
+                mods = get_modifiers(event["modifiers"])
+                typ = TYPLUT[event_type]
+                evt = QtGui.QMouseEvent(typ, pos, pos, btn, btns, mods)
+                QtCore.QCoreApplication.sendEvent(self.gfxView.viewport(), evt)
+                self.request_draw()
+        elif event_type == "wheel":
+            pos = QtCore.QPointF(event["x"], event["y"])
+            pixdel = QtCore.QPoint()
+            scale = -1.0    # map JavaScript wheel to Qt wheel
```

Comment:
From what I've seen, Qt uses steps of 120, versus 96 of JS, so useing `-120/96 == -1.25`  here it should be more precise.

Reply:
I tried out with a mouse wheel. On Qt Desktop, each step generated a delta of 120. But on jupyter_rfb (on Windows Chrome), each step generated a delta of 200 (unscaled). But you are saying that you are getting 96 per step?

Reply:
Yes, I get 96 on Windows Firefox. I assumed that this was more or less standard. Arg. I'll do some more research.

Reply:
I got 100 on Chrome. I wonder why you get 200. Do you happen to be using a high-rez display?

Reply:
Oh yes, I was on a laptop with dpr 2.0.

Reply:
I suppose you "feel" the difference in scrolling too. What about using the touchpad, does it feel more sensitive too?

Reply:
> Oh yes, I was on a laptop with dpr 2.0.

Could you try on Firefox too please? I don't have access to my hidpi screen right now :)

Reply:
On the same Windows laptop with dpr 2.0:
Chrome: 200
Firefox: 96

Reply:
I made an issue, with a little test html to get some more details: https://github.com/vispy/jupyter_rfb/issues/50